SignalServiceKit/src/Storage/TSStorageManager.m

```diff
@@ -198,6 +291,8 @@ - (void)setupDatabaseWithSafeBlockingMigrations:(void (^_Nonnull)())safeBlocking
     [self.database registerExtension:[TSDatabaseSecondaryIndexes registerTimeStampIndex] withName:@"idx"];
     [OWSMessageReceiver syncRegisterDatabaseExtension:self.database];
 
+    setDatabaseInitialized();
```

Comment:
Well, in the absence of finding documentation that can shed light on the issue we've been seeing, this issue only seems to affect sync and not async registrations.  We've always been opening write transactions before the async registrations complete without negative consequences.